ceph: backport installation documentation for Ceph Cluster chart (#8088)
rook/rook: Documentation/helm-ceph-cluster.md

---
title: Ceph Cluster
weight: 10200
indent: true
---

{% include_relative branch.liquid %}

# Ceph Cluster Helm Chart

> Installs a [Ceph](https://ceph.io/) cluster on Rook using the [Helm](https://helm.sh) package manager.

Reviewer: ```suggestion
Install a Rook Ceph cluster using the [Helm](https://helm.sh) package manager.
```

> ## Warning

Reviewer 2: ```suggestion
## Experimental
```

> This chart is released to the community for testing but is considered in development. It is currently only available on the `master` channel.

Reviewer 2: ```suggestion
This chart is released to the community for testing but is considered in experimental. It is currently only available on the `master` channel.
```

> Once released, the repository should be updated to the `release` channel if you wish to switch to the stable branch.

Reviewer 2: Helm will allow changing the channel, right? Hopefully we can confirm this won't be an issue later.
```suggestion
Once released (targeted for v1.7), the repository should be updated to the `release` channel at that time for the stable branch.
```

## Prerequisites

* Kubernetes 1.13+
* Helm 3.x
* Preinstalled Rook Operator. See the [Helm Operator](https://rook.github.io/docs/rook/master/helm-operator.html) topic to install.

## Installing

The `helm install` command deploys rook on the Kubernetes cluster in the default configuration.
The [configuration](#configuration) section lists the parameters that can be configured during installation. It is
recommended that the rook operator be installed into the `rook-ceph` namespace (you will install your clusters into
separate namespaces).

> If the operator was installed in a non-default location, the namespace of the *Rook Operator* installed

Reviewer: ```suggestion
If the operator was installed in a non-default location, the namespace of the **Rook Operator** installed
```

must be set in the `operatorNamespace` variable.

Rook currently publishes builds of this chart to the `master` channel.

### Master Channel

> The master channel is the most recent release of Rook that includes in development features.

Reviewer 2: ```suggestion
The master channel is the most recent release of Rook that includes experimental features.
```

```console

> helm repo add rook-release https://charts.rook.io/master

Reviewer 2: Shall we call it rook-master?
```suggestion
helm repo add rook-master https://charts.rook.io/master
```

helm install --create-namespace --namespace rook-ceph rook-ceph --set operatorNamespace=rook-ceph rook-release/rook-ceph-cluster
```

> ### Development Build

Reviewer 2: Let's remove this section for this experimental documentation, and just focus on the master channel.
